packages/compat/src/one-shot.ts

```diff
@@ -2,27 +2,61 @@ import Plugin from 'broccoli-plugin';
 import { Node } from 'broccoli-node-api';
 import { Builder } from 'broccoli';
 import { copySync } from 'fs-extra';
+import heimdall from 'heimdalljs';
+
+class NerfHeimdallReparentingBuilder extends Builder {
+  /*
+    Replace the code used to track heimdall nodes: https://github.com/broccolijs/broccoli/blob/v3.5.2/lib/builder.ts#L463-L503
+
+    This reduces the amount of memory that these one-shot's create by:
+
+    - Avoiding creating Heimdall nodes for each broccoli plugin
+    - Disabling the "re-parenting" process done by Broccoli builder (which ends up creating **double** the heimdall nodes)
+  */
+  setupHeimdall() {}
+  buildHeimdallTree() {}
+}
 
 // Wraps a broccoli tree such that it (and everything it depends on) will only
 // build a single time.
 export default class OneShot extends Plugin {
-  private builder: Builder;
   private didBuild = false;
 
-  constructor(originalTree: Node) {
+  constructor(private originalTree: Node, private addonName: string) {
     // from broccoli's perspective, we don't depend on any input trees!
     super([], {
+      annotation: `@embroider/compat: ${addonName}`,
       persistentOutput: true,
     });
-    this.builder = new Builder(originalTree);
   }
   async build() {
     if (this.didBuild) {
       return;
     }
-    await this.builder.build();
-    copySync(this.builder.outputPath, this.outputPath, { dereference: true });
-    await this.builder.cleanup();
-    this.didBuild = true;
+
+    // Make a heimdall node so that we know for sure, all nodes created during our
+    // inner builder can be remove easily
+    const oneshotCookie = heimdall.start({
+      name: `@embroider/compat: OneShot (${this.addonName})`,
```

Comment:
OneShot will already be the name of this broccoli-plugin this will look like `OneShot > @embroider/compat: OneShot  addonName`